rust/tools/authorization-logic/src/parsing/astconstructionvisitor.rs

```diff
@@ -154,8 +177,8 @@ fn construct_fact_assertion(ctx: &FactAssertionContext) -> AstAssertion {
 fn construct_hornclause(ctx: &HornClauseAssertionContext) -> AstAssertion {
     let lhs = construct_fact(&ctx.fact().unwrap());
     let mut rhs = Vec::new();
-    for flat_fact_ctx in ctx.flatFact_all() {
-        rhs.push(construct_flat_fact(&flat_fact_ctx));
+    for rvalue_ctx in ctx.rvalue_all() {
+        rhs.push(construct_rvalue(&rvalue_ctx));
     }
```

Comment:
Should this use a higher-ordered function like `map` rather than mutation?

Reply:
fixed